source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -912,6 +912,125 @@ void Graph::updateMaterials(mx::InputPtr input /* = nullptr */, mx::ValuePtr val
     }
 }
 
+// Set the node to display in render view based on selected node or nodegraph
+template <typename T>
+bool Graph::renderEnumInput(mx::InputPtr& input, const mx::UIProperties& ui, T &evalue)
+{
+    
+  // Eg. <input name="subsurface_color" type="color3" value="0.2,0.6,0.05" 
+  //    enum="PRESET1,PRESET2,PRESET3,PRESET4" enumvalues="0.2,0.6,0.05, 0.2,0.7,0.07, 0.2,0.6,0.05, 0.2,0.6,0.05" />
+  std::string enumStr = input->getAttribute(mx::Input::ENUM_ATTRIBUTE);
+  std::string enumValueStr = input->getAttribute(mx::Input::ENUM_VALUES_ATTRIBUTE);
+
+  if (enumStr.empty())
+  {
+    return false;
+  }
+
+  const size_t INVALID_INDEX = std::numeric_limits<size_t>::max();
+  size_t index = INVALID_INDEX;
+  mx::ValuePtr val = input->getValue();
+  evalue = val->asA<T>();
+  std::string enumDelimiter = ",";
+
+  auto splitStringByGroup = [](const std::string& input, int groupSize) {
+    std::vector<std::string> result;
+    std::stringstream ss(input);
+    std::string token;
+    std::string group;
+    int count = 0;
+
+    while (std::getline(ss, token, ',')) {
+        if (!group.empty()) {
+            group += ",";
+        }
+        group += token;
+        count++;
+
+        if (count == groupSize) {
+            result.push_back(group);
+            group.clear();
+            count = 0;
+        }
+    }
+
+    return result;
+ };
+
+  // target enumStr result in eg. ["PRESET1", "PRESET2", "PRESET3", "PRESET4"]
+  // target vec3 enumValues result in eg. ["0.2,0.6,0.05", "0.2,0.7,0.07", "0.2,0.6,0.05", "0.2,0.6,0.05"]
+  std::vector<mx::ValuePtr> enumValues = ui.enumerationValues;
+  std::vector<std::string> enumVec = ui.enumeration;
+
+  //Handling edge cases where enumvalues attribute is empty
+  //1. This can happen if the enum values are not set in the UI properties.
+  //i.e, ui.enumerationValues and ui.enumeration are not populated for non-uniforms. See /MaterialXRender/Util.cpp
+  //
+  //2. or deliberately ignored in the UI properties. 
+  // eg. <input name="filtertype" type="string" value="linear" enum="closest,linear,cubic" uiname="Filter Type" uniform="true" />
+  if (enumValues.empty())
+  {
+
+    //enumvalues can be empty for string types
+    if (enumValueStr.empty() && input->getType() == "string")
+    {
+        enumValueStr = enumStr;
+    }
+
+    if (enumValueStr.empty())
+    {
+        return false;
+    }
+
+    // parse string values from the enum, enumvalues attribute directly and populate enumValues
+    enumVec = mx::splitString(enumStr, enumDelimiter);
+    int groupSize = mx::splitString(enumValueStr, enumDelimiter).size() / enumVec.size();
```

Comment:
Here's the line that likely needs an update for cross-platform compatibility, since the return value of `vector::size` is of type `size_t`.